lib/jekyll/page.rb

```diff
@@ -116,33 +96,37 @@ def url_placeholders
     def process(name)
       self.ext = File.extname(name)
       self.basename = name[0..-ext.length - 1]
+
+      # Invalidate URL
+      @url = nil
+      url
```

Comment:
*Hacky code 3*: This one isn't as bad but since the URL depends on the name, we need to invalidate the url when it changes...